### PR TITLE
fix: resolve contact names and add participant summary context

### LINKED ISSUE #520: Show Apple Contacts names for meeting people
## The situation

Choosing a named person from Apple Contacts can add them to a meeting as only their phone number. This happens when the picker returns a contact snapshot without the name fields Muesli uses for display.

## What works today, and why it is not enough

The selected contact is attached successfully, but Muesli builds the participant directly from the picker snapshot. A phone-only snapshot leaves the meeting header and People list without the contact's saved name, including for contacts stored in configured CardDAV or Exchange accounts.

## Ask

Refetch the selected person as a unified contact from the system Contacts store before building the meeting participant. Use the saved name and existing identity fallbacks when available, and preserve the picker result when Contacts access or lookup fails.

### PR DESCRIPTION
**Pull request by pHequals7**
## Summary

- Port PR #521 by @salmonumbrella onto current `origin/main` while preserving the contributor commit and authorship.
- Refetch selected Apple Contacts as unified contacts so saved display names are recovered across configured Contacts accounts.
- Preserve the picker-derived participant identifier and retain the picker display name when the resolved contact has no usable name.
- Add stored meeting participant names to every meeting summarization path in a distinct, bounded roster section.
- Tell summary models to use the roster for attendance and name context without guessing speaker attribution or participant roles.
- Omit unnamed, email-like, and phone-only display fallbacks, and delimit/escape participant names as untrusted prompt data.
- Add GPT-6 Astra to the applicable ChatGPT and OpenAI model menus for summaries, cleanup, and account-backed computer use.
- Standardize persisted, model-aware Thinking controls across the ChatGPT-account and OpenAI API integrations: meeting summaries and titles, transcript cleanup, and the ChatGPT computer-use planner.
- Keep independent reasoning preferences for summary, cleanup, and computer-use workflows while sharing one capability policy for valid model ranges and defaults.
- Tighten the remote-summary disclosure copy so it remains concise in Settings.

Closes #520

## Review reconciliation

Greptile correctly flagged the first #521 revision for changing a participant identifier after contact resolution, which could create duplicate People entries. The current contributor commit preserves the picker-derived identifier and received Greptile 5/5.

CodeRabbit then identified one remaining edge case: an empty unified contact could overwrite a useful picker name with `Unnamed contact`. This branch fixes that case and adds a regression test.

Claude identified that the stored `displayName` can itself be an email or phone fallback. The roster now filters identity-only values and has regression coverage for full-email and phone-only participants. Claude also noted that the initial character accounting omitted Markdown bullet prefixes; the rendered roster now obeys the 4,000-character cap exactly.

On the integration PR, CodeRabbit requested explicit data delimiters around user-controlled participant names, and Greptile flagged the lack of disclosure for sending previously local-only participant names to hosted or remote summary providers. Both findings are addressed: names are escaped and wrapped in named data delimiters, and the sharing behavior is disclosed in onboarding and Summary Settings.

The previous head also exposed an inconsistent GPT-5.4 Mini computer-use default: that request path still sent `high`, although Mini defaults to `none`. The shared reasoning policy now resolves every supported request path through the selected model's valid range and default, with regression coverage for selected and invalid stored values.

## GPT-5.4 Mini, GPT-6 Astra, and thinking effort

- `gpt-6-astra` is available for ChatGPT-account meeting summaries, transcript cleanup, and computer use, and for OpenAI API summary/cleanup workflows.
- GPT-5.4 Mini exposes None, Low, Medium, High, and Extra High; None is its default.
- Astra exposes Low, Medium, High, Extra High, and Maximum; GPT-5.6 models expose None through Maximum. High remains their default to preserve existing behavior.
- Summary, cleanup, and computer-use preferences are stored independently so tuning one workflow does not silently change another.
- The selected effort is sent as `reasoning.effort` for Responses API calls and `reasoning_effort` for Chat Completions, including generated notes and titles.
- The control is capability-aware: unsupported values fall back to each model's default, and unsupported/custom models do not show it or receive an invalid reasoning field.
- Computer use currently has a ChatGPT-account backend only; its planner now uses the same policy as the ChatGPT and OpenAI API summary/cleanup paths.

## Privacy boundary

Meeting participant records were previously local-only. This change sends name-like stored `displayName` values to the selected meeting-summary backend only when a summary is requested. It does not append participant identifiers or the separate `emailAddress` field. Empty, `Unnamed contact`, email-like, and phone-only values are omitted. The app now states that remote summaries may send transcripts, notes, screen context, and participant names.

## Validation

- PR #521 hosted checks passed on contributor commit `23022c38`, including `ci-gate`, all native shards, release build, DCO, Greptile, CodeRabbit, Socket, and update flow.
- 147 focused reasoning, configuration, model-capability, and request-payload tests passed after the cross-workflow expansion.
- The final exact-tree native suite passed: 2,146 tests in 196 suites.
- Redacted live OpenAI API smoke tests completed successfully against `gpt-5.4-mini-2026-03-17`: explicit Low reasoning returned the expected text, and a planner-shaped request with explicit None reasoning plus a required function tool returned HTTP 200 with a completed `function_call`.
- `swiftc -frontend -parse` passed for all changed Swift source and test files.
- `./scripts/test_classify_changed_files.sh` passed.
- `./scripts/test_ci_test_shards.sh` passed.
- `./scripts/verify_update_flow.sh --skip-dmg` passed.
- `git diff --check` passed.
- Commit `088b9ce8` was built and launched as `/Applications/MuesliDevB.app` using the existing worktree/Xcode caches and a verified pinned LocalVQE runtime. The installed bundle contains the complete runtime and passes deep signature verification.

## Attribution

This integration preserves the signed contributor commit from #521 by @salmonumbrella.

## Contribution certification

- [x] Every non-merge commit has a valid `Signed-off-by` trailer.
- [x] The contribution is submitted under the repository MIT License.
- [x] No new third-party code, models, datasets, media, or other assets are introduced.
- [x] Material AI assistance is disclosed and the resulting changes were reviewed.

### Third-party materials

None.

### AI assistance

OpenAI Codex assisted with review reconciliation, implementation, tests, documentation research, and PR preparation.

## Summary by CodeRabbit

- **New Features**
  - Meeting summaries now include participant names with safer formatting and character limits.
  - Selected contacts are resolved to richer identities while preserving their original identifiers.
  - Added configurable “Thinking” levels for supported summary models, including support for the gpt-6-astra model.
  - Participant names are included when generating, regenerating, or resuming meeting summaries.

- **Documentation**
  - Onboarding and Settings clarify when summary data may leave the Mac for remote providers.

- **Bug Fixes**
  - Improved contact and email handling with graceful fallback when access or lookup fails.

### CODE REVIEW

File: native/MuesliNative/Sources/MuesliNativeApp/MeetingSummaryClient.swift [hunk 1]
```diff
@@ -342,10 +353,20 @@ enum MeetingSummaryClient {
...
         previousMeetingNotes: String? = nil
     ) -> String {
         var prompt = "Meeting title: \(meetingTitle)\n\n"
+        let rosterNames = participantNamesForPrompt(participantNames)
+        logger.info("summary prompt participantNamesIncluded=\(!rosterNames.isEmpty) participantCount=\(rosterNames.count)")
+        fputs("[summary] prompt participantNamesIncluded=\(!rosterNames.isEmpty) participantCount=\(rosterNames.count)\n", stderr)
+        if !rosterNames.isEmpty {
+            prompt += "Meeting participants (roster context only; use these names to understand who attended, but do not infer speaker attribution):\n"
+            prompt += rosterNames.map { "- \($0)" }.joined(separator: "\n")
+            prompt += "\n---\n\n"
```
[the remark this answers is not in the document]
Comment (pHequals7) [reply native/MuesliNative/Sources/MuesliNativeApp/MeetingSummaryClient.swift]: Addressed in d41d032d. Summary Settings and onboarding now disclose that summaries may include participant names and that hosted or remote endpoints send this information off-device.

File: native/MuesliNative/Tests/MuesliTests/MeetingSummaryClientTests.swift [hunk 2]
```diff
@@ -120,6 +121,27 @@ struct MeetingSummaryClientTests {
...
+                "  Priya Shah  ",
+                "Alex\nKim",
+                "PRIYA SHAH",
+                MeetingContactIdentity.unnamedFallback,
+                "  ",
+            ]
+        )
+
+        #expect(prompt.contains("Meeting participants (roster context only"))
+        #expect(prompt.contains("- Priya Shah\n- Alex Kim\n---"))
```
[the remark this answers is not in the document]
Comment (pHequals7) [reply native/MuesliNative/Tests/MuesliTests/MeetingSummaryClientTests.swift]: Addressed in d41d032d: the roster and each participant name now use explicit named data delimiters, delimiter characters are escaped, and an adversarial delimiter-like name is covered by a regression test.

File: native/MuesliNative/Sources/MuesliNativeApp/Models.swift [hunk 3]
```diff
@@ -817,9 +841,47 @@ struct SummaryModelPreset {
     }
 
     static func reasoningEffort(for model: String) -> String? {
+        defaultReasoningEffort(for: model)?.rawValue
+    }
+
+    static func selectableReasoningEfforts(for model: String) -> [SummaryReasoningEffort] {
         switch model.trimmingCharacters(in: .whitespacesAndNewlines) {
+        case "gpt-5.4-mini":
+            return [.off, .low, .medium, .high, .xhigh]
```
[the remark this answers is not in the document]
Comment (pHequals7) [reply native/MuesliNative/Sources/MuesliNativeApp/Models.swift]: Thanks for catching the inconsistent planner behavior. Addressed in 088b9ce8: the approved scope now deliberately standardizes reasoning across computer use, transcript cleanup, and meeting summaries. GPT-5.4 Mini therefore uses its supported default of None in the planner rather than preserving the old omission; computer use has its own persisted setting, and invalid stored choices fall back through the shared model-capability policy. Focused tests cover Mini default None, explicit Extra High, and invalid Maximum fallback, and the final native suite passes all 2,146 tests.

[the remark this answers is not in the document]
Comment (pHequals7) [reply native/MuesliNative/Sources/MuesliNativeApp/Models.swift]: Follow-up live verification: a redacted Responses API request to gpt-5.4-mini-2026-03-17 using the planner-relevant shape (required function tool, parallel_tool_calls false, and explicit reasoning effort none) returned HTTP 200, status completed, with a function_call output. So the standardized explicit None value is accepted by the real backend; this is an intentional contract update rather than an API compatibility regression.
